src/main/java/com/thealgorithms/conversions/TemperatureConversion.java

```diff
@@ -0,0 +1,149 @@
+package com.thealgorithms.conversions;
+
+import java.text.DecimalFormat;
+import java.util.Scanner;
+/**
+ * Convets the temperatures from user to Celsius/Fahrenheit/Kelvin
+ *
+ * @author Tharun S.M.
+ */
+
+public class TemperatureConversion {
+    /*
+     * Object of DecimalFormat Class
+     * to round the output to 2 decimal point value
+     */
+    private static final DecimalFormat rnd = new DecimalFormat("0.00");
+    /*
+     * Main method
+     *
+     * @param args Command line arguments
+     */
+
+    public static void main(String[] args) {
+
+        Scanner sc = new Scanner(System.in);
+
+        // Get the type of temprature input
+        System.out.println("Enter the temperature input type (use numbers)");
+        System.out.println("1 => Celsius");
+        System.out.println("2 => Fahrenheit");
+        System.out.println("3 => Kelvin");
+        int fromType = sc.nextInt();
+
+        // get the actual temprature
+        System.out.println("Enter the temperature :");
+        String a = sc.next();
+        float inTemp = Float.parseFloat(a);
+
+        // get the type you want to convert your temperature to
+        System.out.println("Type you want to convert your temperature to (use numbers)");
+        System.out.println("1 => Celsius");
+        System.out.println("2 => Fahrenheit");
+        System.out.println("3 => Kelvin");
+        int toType = sc.nextInt();
+
+        if (fromType == toType) {
+            System.out.println("Your Temperature is the same !!");
+        } else if (fromType == 1 && toType == 2) {
+            System.out.println("Converted value: " + convertCelsiusToFahrenheit(inTemp) + " °F");
+        } else if (fromType == 1 && toType == 3) {
+            System.out.println("Converted value: " + convertCelsiusToKelvin(inTemp) + " K");
+        } else if (fromType == 2 && toType == 1) {
+            System.out.println("Converted value: " + convertFahrenheitToCelsius(inTemp) + " °C");
+        } else if (fromType == 2 && toType == 3) {
+            System.out.println("Converted value: " + convertFahrenheitToKelvin(inTemp) + " K");
+        } else if (fromType == 3 && toType == 1) {
+            System.out.println("Converted value: " + convertKelvinToCelsius(inTemp) + " °C");
+        } else if (fromType == 3 && toType == 2) {
+            System.out.println("Converted value: " + convertKelvinToFahrenheit(inTemp) + " °F");
+        } else {
+            System.out.println("Please check your input and output types");
+        }
+        sc.close();
+    }
```

Comment:
This is not needed.
```suggestion
```

Reply:
import java.util.Scanner;

public class TemperatureConverter {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        // Get the type of temperature input
        System.out.println("Enter the temperature input type (use numbers)");
        System.out.println("1 => Celsius");
        System.out.println("2 => Fahrenheit");
        System.out.println("3 => Kelvin");
        int fromType = sc.nextInt();

        // Get the actual temperature
        System.out.println("Enter the temperature:");
        float inTemp;
        if (sc.hasNextFloat()) {
            inTemp = sc.nextFloat();
        } else {
            System.out.println("Invalid temperature input. Please enter a valid number.");
            sc.close();
            return;
        }

        // Get the type you want to convert your temperature to
        System.out.println("Enter the temperature output type (use numbers)");
        System.out.println("1 => Celsius");
        System.out.println("2 => Fahrenheit");
        System.out.println("3 => Kelvin");
        int toType = sc.nextInt();

        if (fromType < 1 || fromType > 3 || toType < 1 || toType > 3) {
            System.out.println("Invalid input or output type. Please enter a number between 1 and 3.");
        } else if (fromType == toType) {
            System.out.println("Your Temperature is the same: " + inTemp);
        } else {
            float convertedTemp;
            switch (fromType) {
                case 1:
                    convertedTemp = toType == 2 ? convertCelsiusToFahrenheit(inTemp) : convertCelsiusToKelvin(inTemp);
                    break;
                case 2:
                    convertedTemp = toType == 1 ? convertFahrenheitToCelsius(inTemp) : convertFahrenheitToKelvin(inTemp);
                    break;
                case 3:
                    convertedTemp = toType == 1 ? convertKelvinToCelsius(inTemp) : convertKelvinToFahrenheit(inTemp);
                    break;
                default:
                    convertedTemp = 0;
            }
            System.out.println("Converted value: " + convertedTemp + getUnit(toType));
        }
        sc.close();
    }

    public static float convertCelsiusToFahrenheit(float celsius) {
        return (celsius * 9 / 5) + 32;
    }

    public static float convertCelsiusToKelvin(float celsius) {
        return celsius + 273.15f;
    }

    public static float convertFahrenheitToCelsius(float fahrenheit) {
        return (fahrenheit - 32) * 5 / 9;
    }

    public static float convertFahrenheitToKelvin(float fahrenheit) {
        return (fahrenheit + 459.67f) * 5 / 9;
    }

    public static float convertKelvinToCelsius(float kelvin) {
        return kelvin - 273.15f;
    }

    public static float convertKelvinToFahrenheit(float kelvin) {
        return (kelvin * 9 / 5) - 459.67f;
    }

    public static String getUnit(int type) {
        switch (type) {
            case 1:
                return "C";
            case 2:
                return "F";
            case 3:
                return "K";
            default:
                return "";
        }
    }
}
you can use this code in is more enhanced

Reply:
import java.util.Scanner;

public class TemperatureConverter {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        // Get the type of temperature input
        System.out.println("Enter the temperature input type (use numbers)");
        System.out.println("1 => Celsius");
        System.out.println("2 => Fahrenheit");
        System.out.println("3 => Kelvin");
        int fromType = sc.nextInt();

        // Get the actual temperature
        System.out.println("Enter the temperature:");
        float inTemp;
        if (sc.hasNextFloat()) {
            inTemp = sc.nextFloat();
        } else {
            System.out.println("Invalid temperature input. Please enter a valid number.");
            sc.close();
            return;
        }

        // Get the type you want to convert your temperature to
        System.out.println("Enter the temperature output type (use numbers)");
        System.out.println("1 => Celsius");
        System.out.println("2 => Fahrenheit");
        System.out.println("3 => Kelvin");
        int toType = sc.nextInt();

        if (fromType < 1 || fromType > 3 || toType < 1 || toType > 3) {
            System.out.println("Invalid input or output type. Please enter a number between 1 and 3.");
        } else if (fromType == toType) {
            System.out.println("Your Temperature is the same: " + inTemp);
        } else {
            float convertedTemp;
            switch (fromType) {
                case 1:
                    convertedTemp = toType == 2 ? convertCelsiusToFahrenheit(inTemp) : convertCelsiusToKelvin(inTemp);
                    break;
                case 2:
                    convertedTemp = toType == 1 ? convertFahrenheitToCelsius(inTemp) : convertFahrenheitToKelvin(inTemp);
                    break;
                case 3:
                    convertedTemp = toType == 1 ? convertKelvinToCelsius(inTemp) : convertKelvinToFahrenheit(inTemp);
                    break;
                default:
                    convertedTemp = 0;
            }
            System.out.println("Converted value: " + convertedTemp + getUnit(toType));
        }
        sc.close();
    }

    public static float convertCelsiusToFahrenheit(float celsius) {
        return (celsius * 9 / 5) + 32;
    }

    public static float convertCelsiusToKelvin(float celsius) {
        return celsius + 273.15f;
    }

    public static float convertFahrenheitToCelsius(float fahrenheit) {
        return (fahrenheit - 32) * 5 / 9;
    }

    public static float convertFahrenheitToKelvin(float fahrenheit) {
        return (fahrenheit + 459.67f) * 5 / 9;
    }

    public static float convertKelvinToCelsius(float kelvin) {
        return kelvin - 273.15f;
    }

    public static float convertKelvinToFahrenheit(float kelvin) {
        return (kelvin * 9 / 5) - 459.67f;
    }

    public static String getUnit(int type) {
        switch (type) {
            case 1:
                return "C";
            case 2:
                return "F";
            case 3:
                return "K";
            default:
                return "";
        }
    }
}